pkg/registry/core/namespace/storage/storage.go

```diff
@@ -121,7 +121,7 @@ func (r *REST) Export(ctx context.Context, name string, opts metav1.ExportOption
 }
 
 // Delete enforces life-cycle rules for namespace termination
-func (r *REST) Delete(ctx context.Context, name string, options *metav1.DeleteOptions) (runtime.Object, bool, error) {
+func (r *REST) Delete(ctx context.Context, name string, deleteValidation rest.ValidateObjectFunc, options *metav1.DeleteOptions) (runtime.Object, bool, error) {
```

Comment:
this change means that admission is not actually called for namespaces on the first delete request (since this function does a `r.store.Storage.GuaranteedUpdate` call internally to set the deletion timestamp and namespace finalizer

Reply:
@liggitt Added the validation check for these manual update-on-delete